funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
+        participantList.splice(index, 1);
+        window.Store.addToLocalstorage(queueName, participantList);
+        return true;
+      }
+    }
+  }
+
   $(document).ready(function() {
+    // To disable debug mode
+    Ractive.DEBUG = false;
+
+    var participantTableContent = new Ractive({
+      el: '#participants-table-content',
+      template: '#participant-row',
+      data: {
+        participants: ''
+      }
+    });
+
+    //updateQueue: If participant in "checkin-queue" has already been checked-in then it is removed from checkin queue
+    function updateQueue(queueName) {
+      var participantIDs = window.Queue.peekAllItems(queueName);
+      if(participantIDs) {
+        participantIDs.forEach(function(participantID) {
+          var participant = "[data-pid=" + participantID +"]";
+          if(queueName === "checkin-queue") {
+            if($(participant).hasClass('js-cancel-checkin')) {
+              //Participant has been checked-in so remove from 'checkin-queue' 
+              window.Queue.dequeue("checkin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+          else {
+            if($(participant).hasClass('js-checkin')) {
+              //Participant's check-in has been cancelled so remove from 'cancelcheckin-queue' 
+              window.Queue.dequeue("cancelcheckin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+        });
+      }
+    }
+
+    function getParticipantsData() {
+      $.ajax({
+        type: 'GET',
+        url:  '{{ "{0}event/{1}/participants/json".format(space.url_for(), event.name) }}',
+        timeout: 5000,
+        dataType: 'jsonp',
+        success: function(data) {
+          participantTableContent.set('participants', data.participants).then(function() {
+            if (Modernizr.localstorage) {
+              $('.loader').hide();
+              updateQueue("checkin-queue");
+              updateQueue("cancelcheckin-queue");
+            }
+          });
+        }
+      });
+    }
+
+    function postCheckinStatus(participantID, action) {
+      if(action) {
+        var posturl = "{{ '{0}event/{1}/checkin/'.format(space.url_for(), event.name) }}" + participantID + "?checkin=t";
+      }
+      else {
+        var posturl = "{{ '{0}event/{1}/checkin/'.format(space.url_for(), event.name) }}" + participantID;
+      }
+      $.ajax({
+        type: 'post',
+        url:  posturl,
+        timeout: 5000,
+        dataType: "json",
+        success: function(data) {
+          if(data.checked_in) {
+            window.Queue.dequeue("checkin-queue", data.participant_id);
+          }
+          else {
+            window.Queue.dequeue("cancelcheckin-queue", data.participant_id);
+          }
+        }
+      });
+    }
+
+    function checkQueues() {
```

Comment:
'Checking' makes it seem like a 'read' operation, but what's actually happening here is, the check-in status of the first participant is being posted to the server. So, it looks like the queue is processed, one participant at a time, every 8 seconds? This can be faster. Just send the entire list and let the server do a bulk check-in?